packages/manager/modules/web-paas/src/details/service/add-addon/controller.js

```diff
@@ -0,0 +1,146 @@
+import debounce from 'lodash/debounce';
+import get from 'lodash/get';
+import moment from 'moment';
+
+export default class {
+  /* @ngInject */
+  constructor($q, $scope, $window, Alerter, WebPaas, $translate) {
+    this.$q = $q;
+    this.$scope = $scope;
+    this.$window = $window;
+    this.Alerter = Alerter;
+    this.WebPaas = WebPaas;
+    this.$translate = $translate;
+    this.disabled = false;
+  }
+
+  $onInit() {
+    this.currentMonth = moment().format('MMMM');
+    this.setParameters();
+    this.$scope.$watch(
+      '$ctrl.addon.quantity',
+      debounce(
+        () => (this.addon.quantity > 0 ? this.addStorage() : null),
+        1000,
+      ),
+    );
```

Comment:
Why is this needed ? Can't we use a `onChange` binding on component ?

Reply:
On click of increasing the quannity, we have to fetch the prices from the API, So we had a discussion and decided to use debounce and delay for a second (after user clicks on the button)  before we call the API